robot-core/src/main/java/org/obolibrary/robot/TemplateHelper.java

```diff
@@ -622,6 +944,178 @@ public static OWLEntity getEntity(
     return readXSV(reader, '\t');
   }
 
+  /**
+   * Return true if the template string is valid, false otherwise.
+   *
+   * @param template the template string to check
+   * @return true if valid, false otherwise
+   */
+  public static boolean validateTemplateString(String template) {
+    template = template.trim();
+    if (template.equals("ID")) {
+      return true;
+    } else if (template.equals("LABEL")) {
+      return true;
+    } else if (template.equals("TYPE")) {
+      return true;
+    } else if (template.equals("CLASS_TYPE")) {
+      return true;
+    } else if (template.matches("^(>?C|>{0,2}A[LTI]?|>?P|>?D|>?I|>?R|>?DT) .*")) {
+      return true;
+    } else if (template.startsWith("PROPERTY_TYPE")) {
+      return true;
+    } else if (template.startsWith("CHARACTERISTIC")) {
+      return true;
+    } else if (template.startsWith("INDIVIDUAL_TYPE")) {
+      return true;
+    } else {
+      return template.matches("^(SC|EC|DC|SP|EP|DP|IP|NI|SI|DI|CI|PI|RI|II).*");
```

Comment:
Listing the options explicitly is more clear. But some of these must be followed by a space, and a few are not.

Either way, this overlaps with the previous regex. We should just have one complete regex.

Reply:
Changed to regex for each type of template string (annotations, classes, etc...)